### ISSUE TITLE
Not injected.

### ISSUE THREAD

**Issue by kovan** [barryvdh/laravel-debugbar#232]
Hi. I have tried to use debugbar, but it is not injected in my page.

I have run "composer require barryvdh/laravel-debugbar", and added 'Barryvdh\Debugbar\ServiceProvider' to app.php providers array. I have debug set to true.

The toolbar is not even injected in the code, before the </body> tag.

I'm using Laravel 4.1 and PHP 5.5.9, and I'm available for testing.

Thank you for your time :)

**Comment by illuminate3** [comment]
here"s my log error;
```
[2015-02-08 17:56:12] local.ERROR: Debugbar exception: Route [debugbar.openhandler] not defined.  
```

**Comment by barryvdh** [comment]
Can you clear the laravel route cache (and other caches?)
php artisan route:clear

**Comment by illuminate3** [comment]
That is standard issue protocol for working with Laravel.
I'll go ahead and manually erase all the files in the storage area to see if that makes a difference.

I'll also try this: "You can try disabling all collectors and turn them on one-by-one to see of one is conflicting."

**Comment by headersalreadysent** [comment]
Can you add debugbar provider as first elemnt of provider array in app.php?
9 Şub 2015 16:41 tarihinde "illuminate3" <notifications@github.com> yazdı:

**Comment by illuminate3** [comment]
@headersalreadysent 
Thank you!
That was it almost. I moved it above another ServiceProvider and solved the issue.

This is not a standard package and more of hack to get Whoops back working.
This order solved the injection issue.
```
'Barryvdh\Debugbar\ServiceProvider',
'App\Providers\ErrorServiceProvider',
```
The ErrorSeriviceProvider came from;
https://laracasts.com/discuss/channels/general-discussion/whoops-removed-from-laravel-5?page=4

Interesting that they should conflict.

**Comment by illuminate3** [comment]
hmm, tried to install on a fresh L5

```
[2015-02-10 18:46:43] local.ERROR: Debugbar exception: mkdir(): Permission denied  
```

```
sudo chmod -R 777 storage
```

Will, allow the mkdir command to run ... ugh! Need to change my aliases ...

**Comment by tomzx** [comment]
Is everybody's issue fixed? Can this issue be closed?

**Comment by reigh-saga** [comment]
It is still not working on me. I tried in on my Windows machine and it is working. 
I'm running Laravel 5 and I was having the same problems. It is working on Windows and when I transfer it on my Ubuntu laptop it is now stop working. I already tried the above solution but still it is not working. I checked on the /storage/debugbar/ and the generated file is in this permission: -rw-r--r--.
Anyone?

**Comment by reigh-saga** [comment]
It's now working, it is the apache2 setup/configuration. Need to enable the mod_rewrite and read the .htaccess file.

**Comment by epileftro85** [comment]
i think the issue has its response above, ```php php artisan route:cache``` and put the provider at the top of the list worked for me!

**Comment by MacWorks** [comment]
Not working for me. (Laravel 5.1, Apache with mod_rewrite, PHP 7.0.5) It's not injected into the body. I have it at the top of my service providers, I have published the config, created the storage directory with the correct permissions (it's making JSON files there), I am in debug mode but tried forcing it on just in case. Flipped the include_vendors back and forth, dumbed down my default layout, nothing is logged in Laravel.log.

Do I need to move assets into my public folder or include some script/css tags in my default layout? If so this isn't mentioned in the installation instructions.

Seriously about to give up.

**Comment by MacWorks** [comment]
Found the issue for me. In ServiceProvider.php it's returns when the environment is testing ... which in my case it usually is. I have configured Laravel such that requests on port 8080 use the testing database / testing environment while port 80 do not.

I would prefer a configuration setting for 'disabled_in_testing' or something along those lines.

However I understand that perhaps I'm an edge case here and will have to think about whether I can even modify how I work with my project or just forget about debugbar.
